src/providers/dota_provider.ts

```diff
@@ -21,68 +19,48 @@ export default class DotaProvider extends Provider {
       throw new Error('Could not find Dota 2 game.');
     }
 
-    super(`http://www.dota2.com/patches/`, `Gameplay Patch`, dota);
-
-    const lastPatch = DataManager.getUpdaterData(DotaProvider.key).lastVersion;
-    assertIsDefined(lastPatch);
-    this.lastPatch = lastPatch;
+    super(`http://www.dota2.com/patches/`, `Dota Updates`, dota);
   }
 
-  public async getNotifications(): Promise<Notification[]> {
+  public async getNotifications(updater: Updater, limit?: number): Promise<Notification[]> {
     let notifications: Notification[] = [];
     try {
       const pageDoc = await this.getPatchPage();
       const patchList = await this.getPatchList(pageDoc);
-      let lastPatch = this.lastPatch;
-      const newPatches = [];
-
-      // Discard the old patches
-      // eslint-disable-next-line no-plusplus
-      for (let i = 0; i < patchList.length && patchList[i] !== lastPatch; i++) {
-        newPatches.push(patchList[i]);
-      }
-
-      // Update the last patch version
-      if (newPatches.length > 0) {
-        lastPatch = newPatches[0];
-        this.setLastPatch(lastPatch);
-      }
+      const lastPatch = this.getLastUpdateVersion(updater);
+
+      const newPatches = patchList
+        // Sort ascending (latest patches are last)
+        .sort((a, b) => {
+          if (a < b) {
+            return -1;
+          }
+          if (a > b) {
+            return 1;
+          }
+          return 0;
+        })
```

Comment:
This doesn't work perfectly, for example "9.12c" is larger than "10.1a"

Reply:
Fixed